dist/pythonlibs/testrunner/spawn.py

```diff
@@ -36,6 +40,17 @@ def find_exc_origin(exc_info):
 
 
 def setup_child(timeout=10, spawnclass=pexpect.spawnu, env=None, logfile=None):
+    # Some boards can't be reset after a terminal is open. We currently still
+    # need to reset after opening the terminal because most tests use this as a
+    # way to sync the device and the test. As long as this is still the behavior
+    # we keep both resets.
+    try:
```

Comment:
Maybe move the try/except logic in a separate private function outside setup_child ? This would avoid some code duplication.

Why not only reset here if TESTRUNNER_RESET_SYNC is disabled ? Otherwise the device under test is reset twice is TESTRUNNER_RESET_SYNC is enabled.